vulyk/ext/worksession.py

```diff
@@ -41,12 +41,36 @@ def start_work_session(self, task, user_id):
         By default we use `datetime.now` in the underlying model to save in
         `start_time` field.
 
+        A user should finish a certain task only once, that's why we perform
+        an upsert below.
+
         :param task: Given task
         :type task: vulyk.models.tasks.AbstractTask
         :param user_id: ID of user, who gets new task
         :type user_id: bytes, str, bson.ObjectId
         """
-        self.work_session.objects.create(user=user_id, task=task)
+        self.work_session \
+            .objects(user=user_id,
+                     task=task) \
+            .modify(upsert=True,
+                    set__start_time=datetime.now(),
+                    set__activity=0)
+
+    def record_activity(self, task, user_id, seconds):
+        """
+        Update an activity counter.
+
+        :param task: Current task
+        :type task: vulyk.models.tasks.AbstractTask
+        :param user_id: ID of user, who gets new task
+        :type user_id: bytes, str, bson.ObjectId
+        :param seconds: User was active for
+        :type seconds: int
+        """
+        self.work_session \
+            .objects(user=user_id,
+                     task=task) \
+            .modify(inc__activity=seconds)
```

Comment:
Any extra checks here? Like negative activity or activity longer than now-start of session?